lib/src/debug.dart

```diff
@@ -268,25 +268,72 @@ void dumpLines(List<Chunk> chunks, SplitSet splits) {
 /// Build a string representation of the [piece] tree.
 String pieceTree(Piece piece) {
   var buffer = StringBuffer();
+  _PieceDebugTree(piece).write(buffer, 0);
+  return buffer.toString();
+}
 
-  void traverse(Piece piece) {
-    buffer.write(piece);
+/// A stringified representation of a tree of pieces for debug output.
+class _PieceDebugTree {
+  final String label;
+  final List<_PieceDebugTree> children = [];
 
-    if (piece is! TextPiece) {
-      var children = <Piece>[];
-      piece.forEachChild(children.add);
+  _PieceDebugTree(Piece piece) : label = piece.toString() {
+    piece.forEachChild((child) {
+      children.add(_PieceDebugTree(child));
+    });
+  }
 
-      buffer.write('(');
-      for (var child in children) {
-        if (child != children.first) buffer.write(' ');
-        traverse(child);
-      }
-      buffer.write(')');
+  /// The approximate number of characters needed to print this tree.
```

Comment:
Can you expand this? I was confused by the intention, but I think it would be more clear if it said "needed to print this tree on a single line"